node/web/public/scripts/controllers/checkoutController.js

```diff
@@ -0,0 +1,191 @@
+Hilary.scope('heinz').register({
+    name: 'checkoutController',
+    dependencies: ['newGidgetModule', 'GidgetRoute', 'locale', 'viewEngine','Order', 'Orders', 'Payment', 'jQuery','router'],
+    factory: function ($this, GidgetRoute, locale, viewEngine, Order,Orders, Payment, $,router) {
+        'use strict';
+
+        // temporary put it here, hard code the page to a search result
+        // GET /checkout
+        $this.get['/checkout'] = new GidgetRoute({
+            routeHandler: function (err, req) {
+              console.log("Guest cookie"+ window.document.cookie )
+              if(window.document.cookie == "email=Guest" || window.document.cookie == "")
+              {
+                router.navigate('/Guest');
+              } else {
+
+                  $.ajax({
+                      // consider add email, now hard code on server side first
+                      url: '/api/checkout',
+                      method: 'GET'
+                  }).done(function (data) {
+                      var results = new Orders(data);
+
+                      if (results.orders().length > 0) {
+                          viewEngine.setVM({
+                              template: 't-checkout',
+                              data: results
+                          });
+                      } else {
+                          viewEngine.setVM({
+                              template: 't-order-no-results',
+                              data: { term: "cart" }
+                          });
+                      }
+                  });
+                  $.ajax({
```

Comment:
Just like in your server-side APIs, you are introducing a race condition here. You should next the second ajax call in the `done` function.